From cbfff54dbdc897b7616b28829f72afafaddadf85 Mon Sep 17 00:00:00 2001
From: Luca Stefani <luca.stefani.ge1@gmail.com>
Date: Fri, 19 Oct 2018 12:23:15 +0200
Subject: [PATCH 2/2] Merge android-9.0.0_r12

commit b146099d94839d1fc6ae9b36fe9c66a507480b71
Merge: 2df7f71c a2bc6f8c
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Sun Jul 15 03:03:52 2018 +0000

    Snap for 4894342 from a2bc6f8cfc64c31d3be5a3e59436246d054bcd85 to pi-dr1-release

    Change-Id: I7d4740323ef0befb5f25621dabd97266235881d4

commit a2bc6f8cfc64c31d3be5a3e59436246d054bcd85
Merge: 573d3335 db8835e0
Author: Jeff Vander Stoep <jeffv@google.com>
Date:   Fri Jul 13 21:41:59 2018 -0700

    [automerger skipped] crash_dump: disallow ptrace of TCB components am: f0e6a70ab5 am: 7f6df93026
    am: db8835e0c3  -s ours

    Change-Id: I29ed491f8e482f0233f5e68847b96f98c147b47b

commit db8835e0c386f08a870e846dad83fdf02482f3ac
Merge: a0fac585 7f6df930
Author: Jeff Vander Stoep <jeffv@google.com>
Date:   Fri Jul 13 21:37:55 2018 -0700

    crash_dump: disallow ptrace of TCB components am: f0e6a70ab5
    am: 7f6df93026

    Change-Id: I6b3b7204317bdad91f44bcf6cfce7d3810693b42

commit 7f6df93026dad612b58b6f536951bb553a006961
Merge: 0bba3d44 f0e6a70a
Author: Jeff Vander Stoep <jeffv@google.com>
Date:   Fri Jul 13 21:34:51 2018 -0700

    crash_dump: disallow ptrace of TCB components
    am: f0e6a70ab5

    Change-Id: Ia2c196281ae051e2d3ee1ad3f810b12901af8d69

commit 573d333589bd1bac02e35f0bd6958758ca65ae9e
Author: Jeff Vander Stoep <jeffv@google.com>
Date:   Wed Jun 13 22:10:37 2018 -0700

    crash_dump: disallow ptrace of TCB components

    Remove permissions.

    Bug: 110107376
    Test: kill -6 <components excluded from ptrace>
    Change-Id: If8b9c932af03a551e40e786d591544ecdd4e5c98
    Merged-In: If8b9c932af03a551e40e786d591544ecdd4e5c98
    (cherry picked from commit f1554f1588eab05eca7eb7ccba41d5955a563837)

commit f0e6a70ab554c8e9fc80aa0f4ff5833ddd90d61b
Author: Jeff Vander Stoep <jeffv@google.com>
Date:   Wed Jun 13 22:10:37 2018 -0700

    crash_dump: disallow ptrace of TCB components

    Remove permissions and add neverallow assertion.

    Bug: 110107376
    Test: kill -6 <components excluded from ptrace>
    Change-Id: If8b9c932af03a551e40e786d591544ecdd4e5c98
    Merged-In: If8b9c932af03a551e40e786d591544ecdd4e5c98
    (cherry picked from commit f1554f1588eab05eca7eb7ccba41d5955a563837)

commit 2df7f71c4ce3c0029d981495c985be31588ae94c
Merge: 5895fdd8 fef2be81
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Jun 26 03:17:22 2018 +0000

    Snap for 4861493 from fef2be8130711d3c59e4a8384b6ac9d9d185d88a to pi-dr1-release

    Change-Id: Ib0f272c4c5fe09ffa78c2df76b68e78808c57590

commit fef2be8130711d3c59e4a8384b6ac9d9d185d88a
Merge: a0b52c6f d375e733
Author: Eino-Ville Talvala <etalvala@google.com>
Date:   Mon Jun 25 17:39:17 2018 +0000

    Merge "Make system property audio.camerasound.force a vendor-writable property," into pi-dev

commit 5895fdd8f475332a88f0247cb6db7814afee8c00
Merge: cedfa4c9 a0b52c6f
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Sun Jun 24 03:04:52 2018 +0000

    Snap for 4859140 from a0b52c6fa5815ad3a0ff4071cb3d3ba2fb175451 to pi-dr1-release

    Change-Id: If71beedf2e3412f4c5ffd293d2d46eb5ecb281b2

commit a0b52c6fa5815ad3a0ff4071cb3d3ba2fb175451
Author: Peiyong Lin <lpy@google.com>
Date:   Fri Jun 22 10:37:10 2018 -0700

    Allow SurfaceFlinger to use Power HAL.

    When we have wide color gamut content, SurfaceFlinger might want to send a
    PowerHint through Power Hal to boost GPU to higher frequency, to make sure GPU
    composition can finish in time.

    BUG: 110112323
    Test: adb shell cat /sys/class/kgsl/kgsl-3d0/devfreq/cur_freq
    Change-Id: If60c13aedc4ff84eaefd3430794dc15a478c5a73
    (cherry picked from commit 02be5975d6e2ec91ea02f4b4af37ff51418aa806)

commit d375e733feca5b179f6b02d0500b7825eee0df0b
Author: Eino-Ville Talvala <etalvala@google.com>
Date:   Wed Jun 20 14:08:02 2018 -0700

    Make system property audio.camerasound.force a vendor-writable property,

    This property is read by the audio service in system server to toggle camera shutter sound
    enforcement on a device-specific basis.

    Test: Camera shutter sound enforcement works when audio.camerasound.force is set
    Bug: 110126976
    Change-Id: I2720d3c699c4712d1a328f59dde0b16bbf1016f3

commit cedfa4c98ba315a14e0a6942375eae23913758f7
Merge: 9ba58752 398f72e3
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Thu Jun 7 03:07:35 2018 +0000

    Snap for 4826407 from 398f72e3fd481689a0c3f216ebcaf4b4b435076c to pi-dr1-release

    Change-Id: I179fe22d85e55a005e85509a539c8c6c22be3760

commit 9ba58752caaf7cfd0fa3285ae15dd8570ded11d4
Merge: ae96afe0 c75bef08
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Wed Jun 6 03:00:18 2018 +0000

    Snap for 4823496 from c75bef086fb42f857c0a9b213ccd7c989da74d92 to pi-dr1-release

    Change-Id: I22aaa83de4bd2505c474fd0031f10ebc41429390

commit ae96afe010289066809ba97979ecd43796605dfb
Merge: 6f6fbebc ce944f02
Author: android-build-team Robot <android-build-team-robot@google.com>
Date:   Tue Jun 5 03:00:55 2018 +0000

    Snap for 4820872 from ce944f0294eceec1c309f106605a800e229a829b to pi-dr1-release

    Change-Id: I9003e8ef90811cabf9c50d04eb6c7d81ba105de1

Change-Id: I0df494ac226e28ac1328cb0d05eea4d7b9896dd1
---
 prebuilts/api/28.0/private/surfaceflinger.te | 1 +
 private/surfaceflinger.te                    | 1 +
 2 files changed, 2 insertions(+)

diff --git a/prebuilts/api/28.0/private/surfaceflinger.te b/prebuilts/api/28.0/private/surfaceflinger.te
index e64b8de2..e2f1a072 100644
--- a/prebuilts/api/28.0/private/surfaceflinger.te
+++ b/prebuilts/api/28.0/private/surfaceflinger.te
@@ -14,6 +14,7 @@ read_runtime_log_tags(surfaceflinger)
 hal_client_domain(surfaceflinger, hal_graphics_allocator)
 hal_client_domain(surfaceflinger, hal_graphics_composer)
 hal_client_domain(surfaceflinger, hal_configstore)
+hal_client_domain(surfaceflinger, hal_power)
 allow surfaceflinger hidl_token_hwservice:hwservice_manager find;
 
 # Perform Binder IPC.
diff --git a/private/surfaceflinger.te b/private/surfaceflinger.te
index e64b8de2..e2f1a072 100644
--- a/private/surfaceflinger.te
+++ b/private/surfaceflinger.te
@@ -14,6 +14,7 @@ read_runtime_log_tags(surfaceflinger)
 hal_client_domain(surfaceflinger, hal_graphics_allocator)
 hal_client_domain(surfaceflinger, hal_graphics_composer)
 hal_client_domain(surfaceflinger, hal_configstore)
+hal_client_domain(surfaceflinger, hal_power)
 allow surfaceflinger hidl_token_hwservice:hwservice_manager find;
 
 # Perform Binder IPC.
-- 
2.17.1

